src/main/java/co/cask/cdc/plugins/common/CDCReferencePluginConfig.java

```diff
@@ -0,0 +1,22 @@
+package co.cask.cdc.plugins.common;
+
+import co.cask.cdap.etl.api.validation.InvalidConfigPropertyException;
+import co.cask.hydrator.common.Constants;
+import co.cask.hydrator.common.ReferencePluginConfig;
+
+/**
+ * {@link ReferencePluginConfig} with reference name validation.
+ */
+public class CDCReferencePluginConfig extends ReferencePluginConfig {
+  public CDCReferencePluginConfig(String referenceName) {
+    super(referenceName);
+  }
+
+  public void validate() {
+    if (!containsMacro(Constants.Reference.REFERENCE_NAME) && !References.isValid(referenceName)) {
+      String message = String.format("%s is not a valid id. Allowed characters are letters, numbers, " +
```

Comment:
This message fits better in the References class, as that is where the Pattern is. I think it would be better if References.isValid() was instead References.validate() that throws an exception. This method would then catch that exception and use it's message for the InvalidConfigPropertyException.

Reply:
removed References class and avoided duplication of code from IdUtils